src/crypto/CHIPCryptoPAL.h

```diff
@@ -337,11 +349,52 @@ class P256PublicKey : public ECPKey<P256ECDSASignature>
     uint8_t bytes[kP256_PublicKey_Length];
 };
 
-template <typename PK, typename Secret, typename Sig>
-class ECPKeypair
+template <typename PK, typename Plaintext, typename Serialized, typename Sig, typename Secret>
+class ECKeypair
 {
 public:
-    virtual ~ECPKeypair() {}
+    ECKeypair() { mUsage = SupportedECKeyUsages::SIGNING; }
```

Comment:
This usage change and the added template arguments **may cause major impact** to external users (who have had this stable API for a very long time)

Before this is approved, we need to determine if there is a way to resolve this without impact to users and this needs a  call-out to make sure we don't break assumptions.